File: practica2/c++/progCuenta.cc
#include "progCuenta.h"
#include "read.h"
#include "suma.h"
#include "write.h"
#include "dup.h"
#include "push.h"
#include "jumpif.h"

progCuenta::progCuenta()
: programa::programa(7)
{
	instrucciones[0] = new read();
	instrucciones[1] = new dup();
	instrucciones[2] = new write();
	instrucciones[3] = new push(-1);
	instrucciones[4] = new add();
	instrucciones[5] = new dup();	
	instrucciones[6] = new jumpif(1);
}

File: practica2/java/ProgCuenta.java
public class ProgCuenta extends Programa
{
	public ProgCuenta()
	{
		super(7);
		instrucciones[0] = new Read();
		instrucciones[1] = new Dup();
		instrucciones[2] = new Write();
		instrucciones[3] = new Push(-1);
		instrucciones[4] = new Suma();
		instrucciones[5] = new Dup();
		instrucciones[6] = new Jumpif(1);
	}
}
